src/ipips/ipip-0499.md

```diff
@@ -0,0 +1,107 @@
+
+|                                 | Helia default | Kubo `legacy-cid-v0` (default)    | Storacha default | Kubo `test-cid-v1` | Kubo `test-cid-v1-wide`   | DASL          |
+|---------------------------------|---------------|-----------------------------------|------------------|--------------------|---------------------------|---------------|
+| CID version                     | CIDv1         | CIDv0                             | CIDv1            | CIDv1              | CIDv1                     | CIDv1         |
+| Hash Algo                       | sha-256       | sha-256                           | sha-256          | sha-256            | sha-256                   | sha-256       |
+| Chunk size                      | 1MiB          | 256KiB                            | 1MiB             | 1MiB               | 1MiB                      | not specified |
+| Max links `File` node           | 1024          | 174                               | 1024             | 174                | **1024**                  | not specified |
+| Max links `Directory` node      | ?             | 0                                 | ?                | 0                  | 0                         | ?             |
+| Max fanout `HAMTDirectory` node | 256 blocks    | 256 blocks                        | 256 blocks       | 256 blocks         | **1024**                  | not specified |
+| `HAMTDirectory` threshold       | 256KiB (est)  | 256KiB (est:links[name+cid])      | 1000 **links**   | 256KiB             | **1MiB**                  | not specified |
+| DAG layout                      | balanced      | balanced                          | balanced         | balanced           | balanced                  | not specified |
+| Leaves                          | raw           | raw                               | raw              | raw                | raw                       | not specified |
+| Empty directories               | allowed       | allowed                           | disallowed       | allowed            | allowed                   | not specified |
+
+See related discussion at https://discuss.ipfs.tech/t/should-we-profile-cids/18507/
+
+### User benefit
+
+Reliable, deterministic CIDs allow independent verification of content across tools and ipmlementations.
+
+### Compatibility
+
+Implementations will need to (1) make CID generation settings configurable and (2) support user setting of profiles.
+
+Kubo 0.35 will have [`Import.*` configuration](https://github.com/ipfs/kubo/blob/master/docs/config.md#import) option to control DAG width.
+
+### Security
+
+TODO
+
+### Alternatives
+
+Another approach could be to name profiles based on the key UnixFS/CID parameters, e.g. v1-sha256-balanced-1mib-1024w-raw. This is longer and more convoluted.
+
+
+#### Empty directories
+
+Decision if empty directories should be included is left out of scope.
```

Comment:
@lidel I think we should include this, because the goal of this spec is to make UnixFS encoding deterministic when using the profile, and this stands in the way of this, potentially rendering this whole effort futile if not included. 

I think the profile should dictate whether empty directories are included. This should also be configurable such that you can adjust to your own needs.